### PR TITLE
plan: remove planBuilder.err

### PR DESCRIPTION
**Pull request by zz-jason**
## What have you changed? (mandatory)

This PR removes the `error` field inside the `planBuilder` struct, to:
- force to check the `error` returned by the function of `planBuilder`.
- avoid some potential panic, we didn't check `b.err` in some scenario in the before.

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

- unit test
- explain test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Does this PR need to be added to the release notes? (mandatory)

No

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @XuHuaiyu @coocood PTAL

**Comment by ngaut** [comment]
Well done.

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -2204,45 +2174,42 @@ func extractTableAsNameForUpdate(p LogicalPlan, asNames map[*model.TableInfo][]*
 	}
 }
 
-func (b *planBuilder) buildDelete(delete *ast.DeleteStmt) Plan {
-	if delete.TableHints != nil {
```
Comment (winoros) [inline plan/logical_plan_builder.go]: Why remove this check

Comment (zz-jason) [reply plan/logical_plan_builder.go]: `pushTableHints` will return a `false` if the input table hints is `nil`, this check is unnecessary.

**Pull request merged**